contracts/contracts/coordination/GlobalAllowList.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/access/AccessControlDefaultAdminRules.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./IRitualAuthorizer.sol";
+import "./Coordinator.sol";
+
+contract GlobalAllowList is AccessControlDefaultAdminRules, IRitualAuthorizer {
+    using ECDSA for bytes32;
+
+    Coordinator public coordinator;
+    mapping(uint256 => mapping(address => bool)) public authorizations;
+
+    constructor(
+        Coordinator _coordinator,
+        address _admin
+    ) AccessControlDefaultAdminRules(0, _admin) {
+        coordinator = _coordinator;
+    }
+
+    function setCoordinator(Coordinator _coordinator) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Only admin can set coordinator");
+        coordinator = _coordinator;
+    }
+
+    function isAuthorized(
+        uint32 ritualID,
+        bytes memory evidence,
+        bytes memory digest
+    ) public view override returns(bool) {
+        address recovered_address = keccak256(digest)
+          .toEthSignedMessageHash()
+          .recover(evidence);
+        return authorizations[ritualID][recovered_address];
+    }
+
+    function authorize(uint32 ritualID, address[] calldata addresses) public {
+        require(coordinator.getAuthority(ritualID) == msg.sender,
+            "Only ritual authority is permitted");
+        require(coordinator.isRitualFinalized(ritualID),
+            "Only active rituals can add authorizations");
```

Comment:
Does it make sense to turn these requires into one of those solidity modifiers like `onlyOwner` etc., in this case `onlyAuthorityForActiveRitual` or something like that...

It can be used for authorizing and deauthorizing.

Reply:
✔️